Shared/Samples/Apply scheduled updates to preplanned map area/ApplyScheduledUpdatesToPreplannedMapAreaView.swift

```diff
@@ -0,0 +1,200 @@
+                }
+            }
+            .overlay {
+                // Show a progress view for the offline map sync job that downloads
+                // the available updates.
+                if let progress = model.offlineMapSyncJob?.progress {
+                    VStack(spacing: 16) {
+                        ProgressView(progress)
+                            .progressViewStyle(.linear)
+                            .frame(maxWidth: 200)
+                    }
+                    .padding()
+                    .background(.regularMaterial)
+                    .clipShape(RoundedRectangle(cornerRadius: 15))
+                    .shadow(radius: 3)
+                }
+            }
+            .alert("Scheduled Updates Available", isPresented: $updatesAlertIsPresented) {
+                Button("Cancel", role: .cancel, action: {})
+                Button("Apply") {
+                    Task {
+                        do {
+                            try await model.applyScheduledUpdates()
+                        } catch {
+                            self.error = error
+                        }
+                    }
+                }
+            } message: {
+                // Get the download size for the update.
+                let downloadSizeString = ByteCountFormatter.string(
+                    from: Measurement(
+                        value: Double(model.updatesInfo?.scheduledUpdatesDownloadSize ?? .zero),
+                        unit: UnitInformationStorage.bytes
+                    ),
+                    countStyle: .file
+                )
+                Text("A \(downloadSizeString) update is available. Would you like to apply it?")
+            }
+            .alert("Scheduled Updates Unavailable", isPresented: $noUpdatesAlertIsPresented) {
+                Button("OK", action: {})
+            } message: {
+                Text("There are no updates available.")
+            }
+    }
+    
+    /// Displays different alerts based on the offline map updates info.
+    /// - Parameter info: The updates info from the offline map sync task.
+    private func handleUpdatesInfo(_ info: OfflineMapUpdatesInfo) {
+        switch info.downloadAvailability {
+        case .available:
+            updatesAlertIsPresented = true
+        case .noneAvailable, .indeterminate:
+            noUpdatesAlertIsPresented = true
+        @unknown default:
+            fatalError("Unknown offline map updates info")
+        }
+    }
+}
+
+private extension ApplyScheduledUpdatesToPreplannedMapAreaView {
+    /// The model used to store the geo model and other expensive objects
+    /// used in this view.
+    @MainActor
+    class Model: ObservableObject {
+        /// A map from the mobile map package.
+        @Published private(set) var map = Map()
+        
+        /// The sync job used to apply updates to the offline map.
+        @Published private(set) var offlineMapSyncJob: OfflineMapSyncJob?
```

Comment:
Would it make more sense to just have a reference the `offlineMapSyncJob.progress` if that is all it used for? Not sure...

Reply:
Since that involves creating another property, assigning then dereferencing alongside the job, I'll stick to using the job itself.